tests/test_mem_eff_attention.py

```diff
@@ -48,49 +134,305 @@ def test_key_query_all_ones(device, q_len, kv_len, batch_size, k_len):
     # this should be equivalent to the average over value
     ref = value.mean(1, keepdim=True).expand_as(query)
 
-    assert torch.allclose(out, ref, atol=1e-5)
+    assert_allclose(out, ref, atol=1e-5)
 
 
 @pytest.mark.parametrize("k_len", [5, 6, 32])
 @pytest.mark.parametrize("batch_size", [1, 4])
 @pytest.mark.parametrize("kv_len", [3, 15, 32, 33])
 @pytest.mark.parametrize("q_len", [2, 3, 5])
 @pytest.mark.parametrize("device", _devices)
-def test_logsumexp(device, q_len, kv_len, batch_size, k_len):
+@pytest.mark.parametrize("dtype", [torch.float, torch.half])
+@pytest.mark.parametrize(
+    "op",
+    [
+        xformers.ops.MemoryEfficientAttentionOp,
+        xformers.ops.MemoryEfficientAttentionGenericForwardOp,
+    ],
+)
+def test_logsumexp(
+    device,
+    q_len,
+    kv_len,
+    batch_size,
+    k_len,
+    dtype,
+    op: xformers.ops.MemoryEfficientAttentionOp,
+):
+    scale = 3
```

Comment:
for my understanding, how is this scale chosen ?

Reply:
This is just to stress test a bit more the numerics of MHA. I could have left it as 1, but with larger scales it pushes it harder in the `query @ key.T` part so that we could hit overflows if the softmax is not done properly, for larger `K` dimension.